### PR TITLE
Client: Spider tasks GUI

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
Add a GUI for the client spider tasks. I'll update this PR once that has been merged.

## Related Issues
Depends on https://github.com/zaproxy/zap-extensions/pull/6017

## Checklist
- [ ] Update help
- [ ] Update changelog
- [ ] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [ ] Sign-off commits
- [ ] Squash commits
- [ ] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by psiinon** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [f7c2d0be\-4302\-44b1\-9890\-0fcd240d068d](https://eu.ast.checkmarx.net/projects/0da811fa-6373-42b1-bbb6-d0026285b136/scans?id=f7c2d0be-4302-44b1-9890-0fcd240d068d&branch=psiinon%2Fclient%2Faction-gui)
### No New Or Fixed Issues Found

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: addOns/client/src/main/java/org/zaproxy/addon/client/spider/ClientSpider.java [hunk 1]
```diff
@@ -312,22 +330,45 @@ public void eventReceived(Event event) {
...
+        if (tag != null) {
+            switch (tag) {
+                case "A":
+                    return Constant.messages.getString(
+                            "client.spider.panel.table.details.link",
+                            parameters.get("href"),
+                            parameters.get("text"));
+                case "BUTTON":
+                    return Constant.messages.getString(
+                            "client.spider.panel.table.details.button", parameters.get("text"));
```
Comment (kingthorin) [inline addOns/client/src/main/java/org/zaproxy/addon/client/spider/ClientSpider.java:366]: No need for it to block this, but it seems we should have constants for a bunch of field/item names.

**Review by kingthorin — APPROVED** [review]
LGTM.

**Pull request merged**